ui2/src/pages/network/Clearing.tsx

```diff
@@ -23,13 +28,23 @@ export const ClearingServiceTable: React.FC<Props> = ({ services }) => {
 
   const { contracts: roleOffers, loading: roleOffersLoading } = useStreamQueries(RoleOffer);
   const { contracts: offers, loading: offersLoading } = useStreamQueries(Offer);
+  const { contracts: marketOffers, loading: marketOffersLoading } = useStreamQueries(MarketOffer);
   const { contracts: requests, loading: requestsLoading } = useStreamQueries(Request);
+  const { contracts: marketRequests, loading: marketRequestsLoading } = useStreamQueries(MarketRequest);
+
+  const { contracts: marketServices, loading: marketServicesLoading } = useStreamQueries(
+    MarketService
+  );
 
   const roles = useStreamQueries(Role).contracts;
   const hasRole = roles.length > 0 && roles[0].payload.provider === party;
 
-  const terminateService = async (c: CreateEvent<Service>) => {
-    await ledger.exercise(Service.Terminate, c.contractId, { ctrl: party });
+  const terminateService = async (c: CreateEvent<Service> | CreateEvent<MarketService>) => {
+    if (getTemplateId(c.templateId) === 'Marketplace.Clearing.Service.Service') {
```

Comment:
does anyone know a better way to check/coerce the type here? I couldn't quite figure out how to do it using `typeof` with the union types having type parameters, but it felt like it should be possible.

Reply:
what do u want to check the type of?

Reply:
p.s. it might be nice to have all the right hand side template name strings pulled out as variables or put in an enum or something

Reply:
I have the type of `c` as `CreateEvent<Service> | CreateEvent<MarketService>` so I can use the same function when mixing the two types of offers in the same table here (since they have the same fields and choice structure):  https://github.com/digital-asset/da-marketplace/blob/04b517d2970214ae7f0e33f2c86d5c4bf1791021/ui2/src/pages/network/Clearing.tsx#L160.

So it would be nice to switch on the type of `c` without having to check the equality of the template name as a string.

Reply:
> p.s. it might be nice to have all the right hand side template name strings pulled out as variables or put in an enum or something (this one and the one on line 69 and 82 etc.)

Yup that makes sense, it's too bad the codegen doesn't give anything like that.

Reply:
ohh I see what you are saying now

Reply:
ohh I see what you are saying now. Possibly `instanceof` ? Im doing some googling..